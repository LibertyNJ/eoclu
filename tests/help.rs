//! Integration tests that exercise help functionality.

use util::assert_output_snapshot;

mod util;

#[cfg(target_family="unix")]
mod unix {
    use super::*;

    #[test]
    fn should_display_concise_help_when_passed_short_flag() {
        let mut cmd = util::create_test_command();
        let assert = cmd.arg("-h").assert().success().code(0);
        assert_output_snapshot!(assert.get_output());
    }

    #[test]
    fn should_display_full_help_when_passed_long_flag() {
        let mut cmd = util::create_test_command();
        let assert = cmd.arg("--help").assert().success().code(0);
        assert_output_snapshot!(assert.get_output());
    }
}
