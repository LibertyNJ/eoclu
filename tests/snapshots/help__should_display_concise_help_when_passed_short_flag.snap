---
source: tests/help.rs
expression: snapshot
---
stdout:
A command line utility for EVE Online by CCP Games

Usage: eoclu

Options:
  -h, --help  Print help (see more with '--help')

stderr:
